src/coaster.cpp

```diff
@@ -896,3 +937,68 @@ int CoasterInstance::GetNumberOfCars() const
 	if (number_cars == 0) number_cars = 1;
 	return number_cars;
 }
+
+void CoasterInstance::Load(Loader &ldr)
+{
+	this->RideInstance::Load(ldr);
+
+	this->capacity = (int)ldr.GetLong();
+	this->coaster_length = ldr.GetLong();
+
+	const CoasterType *ct = this->GetCoasterType();
+
+	uint saved_pieces = ldr.GetWord();
+	for (uint i = 0; i < saved_pieces; i++) {
+		uint32 index = ldr.GetLong();
+		this->pieces[i].piece = ct->pieces[index];
+		this->pieces[i].Load(ldr);
+		this->PlaceTrackPieceInWorld(this->pieces[i]);
+	}
+
+	int number_of_trains = (int)ldr.GetLong();
+	if (number_of_trains > 0) this->SetNumberOfTrains(number_of_trains);
```

Comment:
There is a Loader::SetFailMessage(), and a Loader::IsFail() or so, which seems to fit the bill.
It's denoted as for debugging and internal use, I guess technical failure reasons are not really something that a user can understand.  The "IsFail()" would still be usable I think, giving a general message like "Load failed as I don't get what the savegame says"  :p